File: server/src/store/cache/lua/authentication/set_redis_session.lua
local session_key = KEYS[1]
local user_key = KEYS[2]
local session_token_hash = ARGV[1]
local session_id = ARGV[2]
local user_id = ARGV[3]
local expires_at = ARGV[4]
local account_status = ARGV[5]
local role = ARGV[6]
local ttl = tonumber(ARGV[7])

redis.call("HSET", session_key,
    "session_id", session_id,
    "user_id", user_id,
    "expires_at", expires_at,
    "account_status", account_status,
    "role", role
)
redis.call("EXPIRE", session_key, ttl)

redis.call("SADD", user_key, session_token_hash)

return 1

File: server/src/store/cache/authentication/set_redis_session.py
from datetime import datetime
from uuid import UUID

from src.app.config.lua_manager import LuaScriptManager
from src.app.crypto.encryption.hash_blake2s import hash_blake2s
from src.app.logging.logger_setup import get_logger

logger = get_logger(__name__)

async def set_redis_session(
    lua_manager: LuaScriptManager,
    session_token: str,
    session_id: UUID,
    user_id: UUID,
    expires_at: datetime,
    account_status: str = "active",
    role: str = "consumer",
) -> None:
    session_token_hash = hash_blake2s(session_token)
    ttl_seconds = 86400

    session_key = f"session:{session_token_hash}"
    user_key = f"user_sessions:{user_id}"

    try:
        await lua_manager.execute(
            "authentication/set_redis_session",
            [session_key, user_key],
            [
                session_token_hash,
                str(session_id),
                str(user_id),
                expires_at.isoformat(),
                account_status,
                role,
                str(ttl_seconds),
            ]
        )
    except Exception:
        logger.warning("set_redis_session_failed", extra={"user_id": str(user_id)})
